### PR TITLE
feat(config): add `rocksdb.wal_compression` to allow enable wal compression

### PR DESCRIPTION
**Pull request by paragor**
issue: https://github.com/apache/kvrocks/issues/2606

add `rocksdb.wal_compression` to allow enable wal compression
only zstd supported, so field YesOrNo

### CODE REVIEW

File: kvrocks.conf [hunk 1]
```diff
@@ -708,6 +708,15 @@ rocksdb.max_background_flushes -1
...
 rocksdb.max_subcompactions 2
 
+# If enabled WAL records will be compressed before they are written. Only
+# ZSTD (= kZSTD) is supported (until streaming support is adapted for other
+# compression types). Compressed WAL records will be read in supported
+# versions (>= RocksDB 7.4.0 for ZSTD) regardless of this setting when
+# the WAL is read.
+#
+# Default is no
+rocksdb.wal_compression no
```
Comment (PragmaTwice) [inline kvrocks.conf:729]: I think it can be `none` and `zstd` instead of `no` and `yes` to leave some upgrading space.

Comment (paragor) [reply kvrocks.conf:729]: ok, i will fix it

Comment (paragor) [reply kvrocks.conf:729]: fixed

**Review by PragmaTwice — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]

**Comment by mapleFU** [comment]
By the way, may I ask that is this only for write-heavy workload(maybe with long and compress-able key-values)?

**Comment by paragor** [comment]
> By the way, may I ask that is this only for write-heavy workload(maybe with long and compress-able key-values)?

i have a lot key-value (uncompressed size in [256B, 8kB]) with good compression rate (value is json, where sum length of json's keys is bigger then json's values).
i have ~100-200MB/s write, wal and compaction load, so on aws i need use aws ec2 instances with enough cpu (see ec2 instance throughput limits https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-optimized.html). 

finally i have a lot of free cpu. this pull request can help me decrease write load on disk and i can pickup cheaper ec2

**Comment by mapleFU** [comment]
I see, this make sense

**Pull request merged**